### PR TITLE
Add dashboard list resource

### PR DESCRIPTION
**Pull request by nmuesch**
Adds a new dashboard list resource.

### CODE REVIEW

File: datadog/provider.go [hunk 1]
```diff
@@ -32,19 +32,20 @@ func Provider() terraform.ResourceProvider {
 		},
 
 		ResourcesMap: map[string]*schema.Resource{
-			"datadog_downtime":                             resourceDatadogDowntime(),
```
Comment (nmuesch) [inline datadog/provider.go]: Alphabetized while I was here...

**Comment by nmuesch** [comment]
This PR requires - https://github.com/terraform-providers/terraform-provider-datadog/pull/297 to be merged before CI will pass.

File: datadog/resource_datadog_dashboard_list.go [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+	dashItemListV2, err := buildTerraformDashboardListItemsV2(d, completeItemListV2)
+	d.Set("dash_item", dashItemListV2)
+	return err
+}
+
+func resourceDatadogDashboardListDelete(d *schema.ResourceData, meta interface{}) error {
+	id, _ := strconv.Atoi(d.Id())
+	// Deleting the overall List will also take care of deleting its sub elements
+	// Deletion of individual dash items happens in the Update method
+	// Note this doesn't delete the actual dashboards, just removes them from the deleted list
```
Comment (gzussa) [inline datadog/resource_datadog_dashboard_list.go]: I don't understand this note

File: datadog/resource_datadog_dashboard_list.go [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+					Schema: map[string]*schema.Schema{
+						"type": {
+							Type:        schema.TypeString,
+							Required:    true,
+							Description: "The type of these dashboards",
+						},
+						"dash_id": {
+							Type:        schema.TypeString,
+							Required:    true,
+							Description: "The list of dashboard IDs to add",
```
Comment (gzussa) [inline datadog/resource_datadog_dashboard_list.go]: is this a list of ids or just one id?

**Pull request merged**